scripts/tract_indicators.qmd

```diff
@@ -0,0 +1,850 @@
+    mutate(
+      census_tract = paste0(county_fips, tract) %>% str_remove_all("\\."), ## cleaning tract codes for joining
+      subcounty_tract_allocation_factor = subcounty_tract_allocation_factor %>% as.numeric) %>% # converting to numeric
+    # sort by the area of the intersection between the tract and subcounty, then group_by census_tract and select the first row
+    # this selects the subcounty that has the greatest land-area-overlap with the tract
+    group_by(census_tract) %>%
+    arrange(desc(land_area_mi)) %>%
+    summarise(across(everything(), first)) %>%
+    mutate(
+      fips2010 = paste0(county_fips, subcounty_fips),
+      join_key = if_else(county_fips %in% ( income_limits_raw %>% filter(county_count == 1) %>% pull(county_fips) ), county_fips, fips2010),
+      join_key = case_when(
+        ## This county doesn't exist in the income limits data (because the county ceased to exist)
+        ## The current county code for this area is 51019.
+        str_detect(join_key,"^51515") ~ "51019",
+        ## For county 29055 in the income limits data, there are two records with invalid fips2010 values
+        ## We assign to all tracts in the county the mean of these incomes limits (since we can't join to the county subdivision). 
+        str_detect(join_key, "^29055") ~ "29055",
+        T ~ join_key)) %>%
+    select(census_tract, join_key) } else {
+      
+  ## For years 2022-onwards
+  geocorr_xwalk = read_csv(geocorr_xwalk_path) %>%
+    ## cleaning up names
+    rename(
+      county_fips = county, 
+      subcounty_fips = cousub20,
+      county_name = CountyName, 
+      subcounty_name = MCDName,
+      land_area_mi = LandSQMI,
+      subcounty_tract_allocation_factor = afact) %>%
+    slice(2:nrow(.)) %>% # dropping a row with variable labels
+    mutate(
+      census_tract = paste0(county_fips, tract) %>% str_remove_all("\\."), ## cleaning tract codes for joining
+      subcounty_tract_allocation_factor = subcounty_tract_allocation_factor %>% as.numeric) %>%
+    group_by(census_tract) %>%
+    arrange(desc(land_area_mi)) %>%
+    summarise(across(everything(), first)) %>%
+    mutate(
+      fips2010 = paste0(county_fips, subcounty_fips),
+      join_key = if_else(county_fips %in% (income_limits_raw %>% filter(county_count == 1) %>% pull(county_fips) ), county_fips, fips2010)) %>%
+    select(census_tract, join_key)
+  }
+
+## join income limits to the tract-to-subcounty-division crosswalk so that we can
+## then associate hmda data (at the tract level) with applicable income limits
+tract_level_income_limits = geocorr_xwalk %>%
+  ## full join to retain records for American Samoa, PR, USVI, and Guam-based areas that are not included in the crosswalk
+  ## but are included in the income limits file; this also joins some rows for areas in the NE with
+  ## subcounty income limits where census_tract values don't match--we drop these at the end
+  full_join(income_limits_raw, by = "join_key") %>%
```

Comment:
```
TEST_tract_level_income_limits = geocorr_xwalk %>%
      ## full join to retain records for American Samoa, PR, USVI, and Guam-based areas that are not included in the crosswalk
      ## but are included in the income limits file; this also joins some rows for areas in the NE with
      ## subcounty income limits where census_tract values don't match--we drop these at the end
      anti_join(income_limits_raw, by = "join_key")
```
Join works well. All values in `geocorr_xwalk` exist in `income_limits_raw`. `anti_join` dataset has 0 observations.